### PR TITLE
[fleet] upgrade fleet sdk to 2024-05-02-preview

### PR DESCRIPTION
**Pull request by pdaru**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [ ] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
fleet

**Comment by pdaru** [comment]
I'm not sure how/why it made changes to amg dir. Individual commits suggest no changes to those files.

**Comment by pdaru** [comment]
$ azdev style fleet
>
> ===============
> | Style Check |
> ===============
> 
> extension(s): [ fleet ] installed from a wheel may need --include-whl-extensions option

unsure how to resolve this.

@kairu-ms please could you help resolve the 2 issues?

File: src/amg/azext_amg/vendored_sdks/aio/operations/_operations.py
Comment (kairu-ms) [inline src/amg/azext_amg/vendored_sdks/aio/operations/_operations.py:1]: why you changed the amg folder?

**Comment by kairu-ms** [comment]
> I'm not sure how/why it made changes to amg dir. Individual commits suggest no changes to those files.

You can revert back the change in amg folder

**Comment by kairu-ms** [comment]
> $ azdev style fleet
> 
> > # ===============
> > | Style Check |
> > extension(s): [ fleet ] installed from a wheel may need --include-whl-extensions option
> 
> unsure how to resolve this.
> 
> @kairu-ms please could you help resolve the 2 issues?

You should install it from source code instead of wheel. Please uninstall it by `az extension remove --name fleet` first. Then use `azdev extension add fleet` to install from your source code.

**Pull request closed without merging**